addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,408 +24,427 @@
-    			print('Proc feature disabled.')
-    		elseif params[2]:lower() == "pos" then
-   				pposx, pposy = tonumber(params[3]), tonumber(params[4])
-   				initializeproc()
-   			end
-		end
-	end
-end)
-
-
+-------------------------------------------------------------------------------
+-- Parse the chat messages for drops, staggers and time extensions ------------
+-------------------------------------------------------------------------------
 windower.register_event('incoming text',function (original, new, color)
---	print('event_incoming_text function')
-	if timer == 'on' then
-  		a,b,fiend = string.find(original,"%w+'s attack staggers the (%w+)%!")
-   		if fiend == 'fiend' then
-			StaggerCount = StaggerCount + 1
-    		windower.send_command('timers c '..StaggerCount..' 30 down')
-    		return new, color
-    	end
-	end
- 	if tracker == 'on' then
-     	a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+ %w+)..\46")
-     		if item == nil then
-      	 		a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+ %w+)..\46")
-       				if item == nil then
-         				a,b,item = string.find(original,"%w+ obtains an? ..(%w+%-%w+ %w+)..\46")
-          					if item == nil then
-           						a,b,item = string.find(original,"%w+ obtains an? ..(%w+ %w+)..\46")
-         					end
-       				end
-     		end
--- 		a,b,item = string.find(original,"%w+ obtains an? ..(.*)..\46")
- 		if item ~= nil then
- 			item = item:lower()
- 			for i=1, #Currency do
-				if item == Currency[i]:lower() then
-   					Currency[Currency[i]] = Currency[Currency[i]] + 1
-   				end
- 			end
- 			obtainedf()
- 		end
-    	initializebox()
- 	end
- 	return new, color
-end)
-
-function obtainedf()
-	obtained = nil
- 	for i=1,#Currency do
- 		if Currency[Currency[i]] ~= 0 then
-  			if obtained == nil then
-  				obtained = " "
-  			end
-   			obtained = (obtained..Currency[i]..': '..Currency[Currency[i]]..' \n ')
- 		end
- 	end
-end
-
-windower.register_event('zone change', function(id)
-	goodzone = ProcZones:contains(id)
-	if not goodzone then
-		windower.text.set_visibility('proc_box', false)
-	end
+    local time = nil
+    local item = nil
+    local fiend = original:match("%w+'s attack staggers the (%w+)%!")
+    if settings.timer then
+        if fiend == 'fiend' then
+            windower.send_command('timers c "'..current_mob..'" 30 down stun')
+            return new, color
+        end
+    end
+    if string.find(original,"remaining in Dynamis.") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"will be expelled from Dynamis") then
+        obj_time = (tonumber(original:match("%d+")) * 60)
+        end_time = os.time() + obj_time
+    end
+    if string.find(original,"Your stay in Dynamis has been extended by %d+ minutes.") then
+        end_time = end_time + (tonumber(original:match("%d+")) * 60)
+        w.time = os.date('!%H:%M:%S', end_time)
```

Comment:
I notice this case adjusts `w.time`, but the others do not. This one does not adjust `ob_time` though. Is that correct?

Reply:
The 2 cases above are statements of how much time left, not time extensions like this one, so there should be no need to update `w.time`. But it would also mean that there is no need to update the internal objects, which leads me to believe it was made in case the plugin is loaded after entering Dynamis.
Anyways, `obj_time`, `end_time` and `w.time` are all linked, so I'll update them together. I'll also rename `obj_time` to `time_remaining_in_seconds` or something alike.